### ISSUE TITLE
How to compute the loss

### ISSUE THREAD

**Issue by hussein-jafarinia** [ncoudray/DeepPATH#127]
Hi

Thanks for the great work.
I'm trying to implement a "cancer"/"healthy" classifier from your code on PyTorch but there is something I don't understand about training the model itself and I can't say I know what's happening by looking at the code.
To calculate the loss are you giving the label of the slide to every patch and then compute loss and train with that assuming to be true (although for some patches  if the slide is cancer the label "cancer" would be definitely be wrong)? Or you're just aggregating the probabilities for every patch (batch by batch) and after the final step you compute the loss between Softmax of the aggregated probabilities and the entire slide's label (something like multiple instance learning)?
I've already done the former and the model is not being trained (It just gives all patches label "cancer")

Thank you

**Comment by ncoudray** [comment]
Hi - The classifier and loss are done on a patch bases. So there is no aggregation done during the training.

HTH,
Best
